### PR TITLE
Feature/add artifact registry

### PR DESCRIPTION
**Pull request by jacobcrawford**
**Description of your changes:**
I have moved the artifact-registry to the mono repo. The tests are added as workflows that only trigger when changes happen in the folder containing this helm chart. Migrating the tests to fit the generic test workflow will be done in the future

Checklist:

* [ ] I have bumped the chart version
* [x] I have documented my changes if this is needed
* [x] I have described my changes in the "description of your changes" field
* [ ] I have tested this change on a running cluster and the Argocd dashboard shows healthy and synced
* [x] I have created the necessary tests in the chart, if they are possible/necessary
* [ ] I have squashed commits if necessary

### CODE REVIEW

**Review by klausenbusk — COMMENTED** [review]
How is files uploaded to the helm server?

**Comment by jacobcrawford** [comment]
At the moment they are just served. This means that charts are put on the server through a PVC mount.

File: charts/artifact-registry/chart/templates/cm-nginx.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        worker_connections  1024;
+    }
+    http {
+      server {
+          listen       80;
+          location / {
+              root   /charts;
+              index  index.yaml;
+          }
+      }
```
Comment (klausenbusk) [inline charts/artifact-registry/chart/templates/cm-nginx.yaml:17]: The indentation is off:
```suggestion
    http {
        server {
            listen       80;
            location / {
                root   /charts;
                index  index.yaml;
            }
        }
```

File: charts/artifact-registry/chart/templates/deployment-image-registry.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            - -v
+            - "localhost"
+            - "443"
+        env:
+        - name: REGISTRY_HTTP_ADDR
+          value: 0.0.0.0:443
+        - name: REGISTRY_HTTP_TLS_CERTIFICATE
+          value: /certs/domain.crt
+        - name: REGISTRY_HTTP_TLS_KEY
+          value: /certs/domain.key
```
Comment (klausenbusk) [inline charts/artifact-registry/chart/templates/deployment-image-registry.yaml:42]: How are these added to the volume? The volume is currently a `emptyDir`.

Do we need TLS or can we can just KISS and don't use TLS for now?

File: charts/artifact-registry/README.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# services-artifact-registry
+A registry setup for both helm charts and container images.
+
+
+To upload the packaged helm chart `chart-0.1.0.tgz` to the endpoint with curl:   
+`curl <IP>:<PORT>/upload -F 'data=@chart-0.1.0.tgz'`
```
Comment (klausenbusk) [inline charts/artifact-registry/README.md:6]: Not relevant anymore.
```suggestion
```

File: charts/artifact-registry/chart/templates/deployment-image-registry.yaml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        resources:
+        {{- toYaml .resources | nindent 12}}
+        readinessProbe:
+          exec:
+            command:
+            - nc
+            - -z
+            - -v
+            - "localhost"
+            - "{{.service.port}}"
```
Comment (klausenbusk) [inline charts/artifact-registry/chart/templates/deployment-image-registry.yaml:35]: Can we just use a normal HTTP probe here? Ex:
```suggestion
          httpGet:
            port: {{.service.port}}
```

File: charts/artifact-registry/chart/templates/deployment-image-registry.yaml (hunk 2 above)
Comment (jacobcrawford) [reply charts/artifact-registry/chart/templates/deployment-image-registry.yaml:43]: I have disabled TLS as default.

**Review by klausenbusk — APPROVED** [review]
LGTM 🎉

**Pull request merged**